### PR TITLE
Engg:: Add testing guide for engineering diffraction interface

### LINKED ISSUE #17144: Testing guide for Engg Diffraction GUI
Write a guide to test manually all the tabs of the EnggDiffraction GUI.
This is complementary to the user documentation and needs to specify the following details, so that anyone could apply the guide to test the interface:
- Inputs and files required, and where to get them.
- Sequence of actions, step by step.

It should cover all the standard usage of the interface, and it could also provide hints for negative tests (for example, leave a particular field blank, push buttons in an uncommon order, etc.).

One or two screenshots wouldn't hurt if that helps the reader, but don't spend too much time on this.

For the test data we can use the neutron training course datasets 2015 and 2016.

### PR DESCRIPTION
**Pull request by DavidFair**
Description of work.
Adds a testing guide for developers who are unfamiliar with Engineering Diffraction to walk them through a typical workflow. Also lists some ideas for negative testing which can be expanded by future developers.

**To test:**
Docs tests pass.
Documentation is clear and coherent esp. to a developer who has no previous experience using this interface 

Fixes #17144 .
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/interfaces/Engineering_Diffraction_TestGuide.rst [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+interface. Whilst users may benefit from the additional documentation provided in this document
+it is recommended they read :ref:`Engineering_Diffraction-ref` instead which is tailored towards users.
+
+The files used in the following examples are found within System test files within 
+`<MantidBuildDir>/ExternalData/Testing/Data/SystemTest` after building the *SystemTestData* target.
+
+Overview
+^^^^^^^^
+The Engineering Diffraction interface allows scientists using the EnginX instrument to interactively
+process their data. There are 5 tabs from which are ordered according to the main steps performed.
```
Comment (DimitarTasev) [inline docs/source/interfaces/Engineering_Diffraction_TestGuide.rst]: `from which are`

**Comment by DimitarTasev** [comment]
Other than the comment, looks good. I went through `Engineering_Diffraction` first so I have notes for that if you want them.

For the testing itself, on most `Browse` windows you can select multiple files (intentional maybe?), but you can select lots of files and then when you press any button to perform an action it runs the algorithms on all of them.

Also Mantid doesn't crash from wrong input files in most places, but the `Fitting` tab seems to crash Mantid when run with an invalid `.nxs` file

**Comment by DavidFair** [comment]
@DimitarTasev I've added the changes we discussed plus highlighting for the RB number - could you try it out and let me know if it is a bit more clear for you?

**Comment by DimitarTasev** [comment]
The warning is clear and visible:


The documentation changes look good.

Everything looks good, :shipit:

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp [hunk 2]
```diff
@@ -658,6 +658,15 @@ void EnggDiffractionViewQtGUI::enableTabs(bool enable) {
...
 }
 
+void EnggDiffractionViewQtGUI::highlightRbNumber(bool isValid) {
+	if (!isValid) {
+		m_ui.label_RBNumber->setStyleSheet("background-color: red; color : white;");
+	}
+	else {
+		m_ui.label_RBNumber->setStyleSheet("background-color: white");
+	}
+}
```
Comment (FedeMPouzols) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp]: 👍

**Comment by FedeMPouzols** [comment]
Geat to see this in place! Yes, the capability to process múltiple files is a feature, and an important one, as data for a single experiment/analysis are usually split in several "run" files. If that's not obvious maybe it should be emphasized somewhere very visible.

**Pull request merged**